controller/ResultController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoLtiConsumer\controller;
+
+use GuzzleHttp\Psr7\Response;
+use oat\oatbox\action\Action;
+use oat\oatbox\log\LoggerAwareTrait;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+use common_exception_BadRequest;
+use common_report_Report as Report;
+use oat\taoResultServer\models\classes\ResultService;
+use oat\taoLtiConsumer\model\classes\ResultService as LtiResultService;
+
+class ResultController extends \tao_actions_CommonModule
+{
+    use ServiceLocatorAwareTrait;
+
+    private $resultService;
+
+    public function __construct(LtiResultService $resultService)
+    {
+        $this->resultService = $resultService;
+    }
+
+    public function actionResultScore()
+    {
+        $payload = $this->getRequestParameter('payload');
+        $this->manageResult($payload);
+    }
+
+    /**
+     * Create a list or a list element
+     * @return Response
+     * @throws common_exception_BadRequest
+     */
+    public function manageResult($payload)
+    {
+        if (!$this->isXmlHttpRequest()) {
+            // throw new common_exception_BadRequest('wrong request mode');
+        }
+
+        $result = $this->resultService->loadPayload($payload);
```

Comment:
$result should be the content of `$this->resultService->getResult();` directly I think.
This would make more sense to maybe call the method `parseXml` or `parsePayload`, which can return an array with the parameters extracted from the payload or an exception if something is wrong, what do you think?